File: frontend/src/types/employee.ts
export interface Employee {
  id: number;
  name: string;
  position: string;
  directReports: number[];
  hireDate: string;
  active: boolean;
}

export interface EmployeeResponse {
  employee: Employee;
  directReports: string[]; // Array of direct report names
}

export interface CreateEmployeeRequest {
  name: string;
  position: string;
  directReports: number[];
  managerId?: number;
} 

File: flywire-exercise-main/src/main/java/com/flywire/exercise/model/Employee.java
package com.flywire.exercise.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Employee {
    private int id;
    private String name;
    private String position;
    private List<Integer> directReports;
    private String hireDate; // keep as String for JSON mapping
    private boolean active;
}
